src/react-apps/applications/service-development/src/features/handleMergeConflict/components/HandleMergeConflictDiscardChanges.tsx

```diff
@@ -0,0 +1,116 @@
+import { createMuiTheme, createStyles, withStyles, WithStyles } from '@material-ui/core/styles';
+import * as React from 'react';
+import AltinnButton from '../../../../../shared/src/components/AltinnButton';
+import AltinnPopover from '../../../../../shared/src/components/AltinnPopover';
+import altinnTheme from '../../../../../shared/src/theme/altinnStudioTheme';
+import { getLanguageFromKey } from '../../../../../shared/src/utils/language';
+import { get } from '../../../../../shared/src/utils/networking';
+const theme = createMuiTheme(altinnTheme);
+
+const styles = () => createStyles({
+  textDisabled: {
+    color: theme.altinnPalette.primary.grey,
+  },
+  input: {
+    marginRight: 49,
+  },
+});
+
+export interface IHandleMergeConflictDiscardChangesProps extends WithStyles<typeof styles> {
+  disabled?: boolean;
+  language: any;
+}
+
+export interface IHandleMergeConflictDiscardChangesState {
+  anchorEl: any;
+  popoverState: any;
+}
+
+const initialPopoverState = {
+  descriptionText: '',
+  isLoading: false,
+  shouldShowDoneIcon: false,
+  btnText: 'OK',
+  shouldShowCommitBox: false,
+  btnMethod: '',
+  btnCancelText: '',
+};
+
+class HandleMergeConflictDiscardChanges extends
+  React.Component<IHandleMergeConflictDiscardChangesProps, IHandleMergeConflictDiscardChangesState> {
+
+  constructor(_props: IHandleMergeConflictDiscardChangesProps) {
+    super(_props);
+    this.state = {
+      anchorEl: null,
+      popoverState: initialPopoverState,
+    };
+  }
+
+  public discardChangesPopover = (event: any) => {
+    this.setState({
+      anchorEl: event.currentTarget,
+      popoverState: { // TODO: Immutability-helper
+        ...this.state.popoverState,
+        btnMethod: this.discardChangesConfirmed,
+        btnText: getLanguageFromKey('handle_merge_conflict.discard_changes_button_confirm',
+          this.props.language),
+        descriptionText: getLanguageFromKey('handle_merge_conflict.discard_changes_message',
+          this.props.language),
+        btnCancelText: getLanguageFromKey('handle_merge_conflict.discard_changes_button_cancel',
+          this.props.language),
+      },
+    });
+  }
+
+  // TODO: Add a spinner
```

Comment:
I have added an altinnSpinner in shared, check it that can be used

Reply:
Noted in outstanding issue #918 will be solved.